homeassistant/components/nissan_leaf.py

```diff
@@ -0,0 +1,467 @@
+                                     min=MIN_UPDATE_INTERVAL))),
+        vol.Optional(CONF_CLIMATE_INTERVAL,
+                     default=DEFAULT_CLIMATE_INTERVAL): (
+                         vol.All(cv.time_period,
+                                 vol.Clamp(
+                                     min=MIN_UPDATE_INTERVAL))),
+        vol.Optional(CONF_FORCE_MILES, default=False): cv.boolean
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+LEAF_COMPONENTS = [
+    'sensor', 'switch', 'binary_sensor', 'device_tracker'
+]
+
+SIGNAL_UPDATE_LEAF = 'nissan_leaf_update'
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set-up the Nissan Leaf component."""
+    import pycarwings2
+
+    async def async_setup_leaf(car_config):
+        """Set-up a car."""
+        _LOGGER.debug("Logging into You+Nissan...")
+
+        username = car_config[CONF_USERNAME]
+        password = car_config[CONF_PASSWORD]
+        region = car_config[CONF_REGION]
+        leaf = None
+
+        async def leaf_login():
+            nonlocal leaf
+            sess = pycarwings2.Session(username, password, region)
+            leaf = sess.get_leaf()
+
+        try:
+            # this might need to be made async (somehow) causes
+            # homeassistant to be slow to start
+            await hass.async_add_job(leaf_login)
+        except(RuntimeError, urllib.error.HTTPError):
+            _LOGGER.error(
+                "Unable to connect to Nissan Connect with "
+                "username and password")
+            return False
+        except KeyError:
+            _LOGGER.error(
+                "Unable to fetch car details..."
+                " do you actually have a Leaf connected to your account?")
+            return False
+        except pycarwings2.CarwingsError:
+            _LOGGER.error(
+                "An unknown error occurred while connecting to Nissan: %s",
+                sys.exc_info()[0])
+            return False
+
+        _LOGGER.info("Successfully logged in and fetched Leaf info")
+        _LOGGER.info(
+            "WARNING: This may poll your Leaf too often, and drain the 12V"
+            " battery.  If you drain your cars 12V battery it WILL NOT START"
+            " as the drive train battery won't connect."
+            " Don't set the intervals too low.")
+
+        data_store = LeafDataStore(leaf, hass, car_config)
+        hass.data[DATA_LEAF][leaf.vin] = data_store
+
+        for component in LEAF_COMPONENTS:
+            if (component != 'device_tracker' or
+                    car_config[CONF_NCONNECT] is True):
+                load_platform(hass, component, DOMAIN, {}, car_config)
+
+        async_track_point_in_utc_time(hass, data_store.async_update_data,
+                                      utcnow() + INITIAL_UPDATE)
+
+    hass.data[DATA_LEAF] = {}
+    tasks = [async_setup_leaf(car) for car in config[DOMAIN]]
+    if tasks:
+        yield from asyncio.wait(tasks, loop=hass.loop)
```

Comment:
Correct syntax (Python 3.5+):
```suggestion
        await asyncio.wait(tasks, loop=hass.loop)
```

Reply:
Fixed with commit https://github.com/home-assistant/home-assistant/pull/19786/commits/60b91781f18dc7c9c889a49902af8fb60aaba196